### PR TITLE
[misc] [doc] Rename some profiler APIs and add docstring, mark old names as deprecated

### PR DESCRIPTION
**Pull request by yolo2themoon**
Related issue = #2579

Add docstring for:
```
    kernel_profiler_print() 
    kernel_profiler_clear()
    kernel_profiler_total_time()
    memory_profiler_print()
    query_kernel_profiler()
```

Rename:
```
    kernel_profiler_print() >>> print_kernel_profile_info()
    kernel_profiler_clear() >>> clear_kernel_profile_info()
    memory_profiler_print() >>> print_memory_profile_info()
    query_kernel_profiler() >>> query_kernel_profile_info()
```
(and mark old names as deprecated)

misc:
Execute sync() before clear() & query() in kernel_profiler APIs

### CODE REVIEW

**Review by k-ye — COMMENTED** [review]
Thanks! It seems that `print_kernel_profile_info()` is more consistent with `print_profile_info`. WDYT?

**Comment by yolo2themoon** [comment]
'profile' is a more accurate word than 'profiler', thanks!

File: taichi/program/program.h [hunk 1]
```diff
@@ -143,14 +143,14 @@ class Program {
...
-  KernelProfilerQueryResult query_kernel_profiler(const std::string &name) {
+  KernelProfilerQueryResult query_kernel_profile(const std::string &name) {
     KernelProfilerQueryResult query_result;
     profiler->query(name, query_result.counter, query_result.min,
                     query_result.max, query_result.avg);
     return query_result;
   }
 
-  void clear_kernel_profiler() {
+  void clear_kernel_profile() {
```
Comment (k-ye) [inline taichi/program/program.h:153]: info?

File: taichi/program/program.h [hunk 2]
```diff
@@ -143,14 +143,14 @@ class Program {
     double avg{0.0};
   };
 
-  KernelProfilerQueryResult query_kernel_profiler(const std::string &name) {
+  KernelProfilerQueryResult query_kernel_profile(const std::string &name) {
```
Comment (k-ye) [inline taichi/program/program.h:146]: _info?

**Review by k-ye — APPROVED** [review]
LGTM & nits!

File: taichi/program/program.h (hunk 1 above)
Comment (yolo2themoon) [reply taichi/program/program.h]: d8ccdb7332385d18e416150628766363d12610ca

File: taichi/program/program.h (hunk 2 above)
Comment (yolo2themoon) [reply taichi/program/program.h]: d8ccdb7332385d18e416150628766363d12610ca

**Comment by taichi-gardener** [comment]
|Baseline.x64|||||||||||
|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
|**data_size**|4.0KB|16.0KB|64.0KB|256.0KB|1.0MB|4.0MB|16.0MB|64.0MB|256.0MB|1.0GB|
|**repeat**|100|100|100|100|100|100|10|10|10|10|
|kernel elapsed time(ms)||||||||||
|fill.i32|0.011|0.0129|0.0279|0.067|0.226|0.864|3.48|13.8001|55.3091|220.2568|
|fill.i64|0.011|0.011|0.0191|0.0401|0.1159|0.4351|1.7669|7.108|28.125|111.9401|
|fill.f32|0.0091|0.0141|0.025|0.0651|0.2291|0.8719|3.4711|13.979|55.6989|221.642|
|fill.f64|0.011|0.0119|0.02|0.0401|0.119|0.4389|1.786|7.1681|28.403|113.1182|
|kernel elapsed time(ms)||||||||||
|reduction.i32|0.0119|0.016|0.0298|0.077|0.2611|1.008|4.0431|16.221|64.631|258.4541|
|reduction.i64|0.011|0.0119|0.021|0.0439|0.139|0.5071|2.07|8.2691|33.015|131.5749|
|reduction.f32|0.01|0.0191|0.041|0.1411|0.5748|2.2149|8.3671|31.9262|72.329|289.0401|
|reduction.f64|0.01|0.0141|0.0269|0.0739|0.273|0.9961|4.4911|16.037|70.2939|263.1462|

|Baseline.cuda|||||||||||
|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
|**data_size**|4.0KB|16.0KB|64.0KB|256.0KB|1.0MB|4.0MB|16.0MB|64.0MB|256.0MB|1.0GB|
|**repeat**|1000|1000|1000|1000|1000|1000|100|100|100|100|
|kernel elapsed time(ms)||||||||||
|fill.i32|0.0064|0.0064|0.0065|0.0065|0.0073|0.0204|0.0883|0.3652|1.4906|5.8986|
|fill.i64|0.0047|0.0047|0.0065|0.0065|0.0069|0.0187|0.0599|0.26|1.1073|4.352|
|fill.f32|0.0047|0.0064|0.0064|0.0065|0.0072|0.0187|0.0825|0.3592|1.4933|5.8572|
|fill.f64|0.0047|0.0064|0.0065|0.0065|0.0069|0.0187|0.0645|0.2536|1.1354|4.3663|
|kernel elapsed time(ms)||||||||||
|reduction.i32|0.0085|0.0085|0.0084|0.0088|0.0105|0.0228|0.0684|0.2633|1.1576|4.9971|
|reduction.i64|0.07|0.1006|0.0758|0.0747|0.1011|0.086|0.0979|0.2606|0.8931|3.4756|
|reduction.f32|0.0104|0.0146|0.0146|0.0146|0.0146|0.0237|0.07|0.2672|1.194|5.119|
|reduction.f64|0.2028|0.2028|0.2028|0.2028|0.2028|0.2057|0.215|0.3883|1.0682|3.4981|

**Pull request merged**